src/client/I2PService.h

```diff
@@ -199,9 +199,7 @@ class TCPIPAcceptor : public I2PService {
   virtual std::shared_ptr<I2PServiceHandler> CreateHandler(
       std::shared_ptr<boost::asio::ip::tcp::socket> socket) = 0;
 
-  virtual std::string GetName() {
-    return "generic TCP/IP accepting daemon";
-  }
+  std::string GetName() const { return "generic TCP/IP accepting daemon"; }
```

Comment:
If the string has a name, it's not longer temporary (it's an lvalue).
Returning `const std::string&` used to be better, although its importance has somewhat decreased because of move constructors.

I do agree that there is a performance cost of creating a `std::string` object, though. But we should accept this cost, for many good reasons. Firstly, because the cost is likely to be minimal. Bound checking on a `const char*` is terrible and it has been the cause of many security issues. Also, raw pointers in general complicate memory management (although in this case, that's not applicable). Optimizations are premature and even unnecessary, and yes, they are even a bit evil.
Optimization of C++ code at such a low level is far too complex for ordinary human being such as us. We should improve our logic and algorithms instead, because that's where the real improvements can be made.

That said, I do appreciate that you are looking into all of this old crappy code. But at this stage of the project it would be more interesting if you could search for bugs and optimizations at the logical level. In fact, that'd be awesome for us.